### PR TITLE
Update CI and Readme

### LINKED ISSUE #6: update coockiecutter template
Assigning this to me

### PR DESCRIPTION
**Pull request by devngc**
- Improved instruction in readme
- Added the docker build and semantic release to CI
- Added releaserc to the template

resolves #6

### CODE REVIEW

**Comment by devngc** [comment]
You might ask me why I have a step to generate app name. Firstly, It uses the repo name as the app name. This will be useful if the user wants to change the name of the app. They can just rename the repository rather than generate the whole new template using the `pollination-apps new` command. Secondly, it makes sure the app name is lowercase.

File: pollination_apps/template/assets/{{cookiecutter.project_slug}}/README.md [hunk 1]
```diff
@@ -46,8 +46,29 @@ You need to install Docker on your machine in order to be able to run this comma
...
+   - `chore`: Other changes that don't modify src/test/
+   - `build`: Changes that affect the build system or external dependencies (example: changing the version of a dependency)
+   - `ci`: Changes to our CI or CD pipelines
+
+   Examples of commit messages:
+
+   - fix: Remove unused imports
+   - feat: Add capability to use analysis period
+
+   **Note** that the commit messages with only `fix` and `feat` type will trigger a deployment to Pollination.
```
Comment (mostaphaRoudsari) [inline pollination_apps/template/assets/{{cookiecutter.project_slug}}/README.md:71]: This is really complicated for a typical user. If you want to change the CI to use the semantic release, then it should be an option. We will have 3 options: no CI, manual versioning, automated versioning

**Review by mostaphaRoudsari — CHANGES_REQUESTED** [review]

File: pollination_apps/template/assets/{{cookiecutter.project_slug}}/README.md (hunk 1 above)
Comment (devngc) [reply pollination_apps/template/assets/{{cookiecutter.project_slug}}/README.md:71]: @mostaphaRoudsari, if a user wants to use their own version numbers then why would they go the GitHub route? They would simply use the command line.

Comment (mostaphaRoudsari) [reply pollination_apps/template/assets/{{cookiecutter.project_slug}}/README.md:71]: I disagree. Setting up the CI to push the app to Pollination once a new release is created on GitHub is an improvement and I think we should consider it as an option.

**Review by mostaphaRoudsari — APPROVED** [review]
Thanks, @devngc! LGTM.

**Pull request merged**